src/tiled/documentmanager.cpp

```diff
@@ -290,11 +290,26 @@ bool DocumentManager::switchToDocument(Document *document)
  *
  * If the given map document is not open yet, a tab will be created for it.
  */
-void DocumentManager::switchToDocument(MapDocument *mapDocument, QPointF viewCenter, qreal scale)
+void DocumentManager::switchToDocument(MapDocument *mapDocument,
+                                       QPointF viewCenter,
+                                       qreal scale,
+                                       bool tryPreserveLayer)
 {
+    const QString *currentLayerName = nullptr;
+    if (tryPreserveLayer) {
+        Document *currentDocument = mMapEditor->currentDocument();
+        if (auto currentMapDocument = qobject_cast<MapDocument*>(currentDocument)) {
```

Comment:
Since this keeping of the current layer is a very world-specific feature, I would suggest to put the logic entirely in `MapItem::mouseReleaseEvent` for now. That way, we avoid the boolean parameter as well.